### ISSUE TITLE
[QUESTION] [Multi-label classification] Does an unbalanced class distribution cause problems during training?

### ISSUE THREAD

**Issue by enescicek** [lightly-ai/lightly-train#964]
I want to train a multi-class image classification model using LightlyTrain. My dataset will contain only the following 11 class/label combinations.

### Data Distribution

| Class/label combination | Number of images |
|---|---:|
| Class3 | 1,756 |
| Class2 | 1,441 |
| Class9 | 1,169 |
| Class4 | 327 |
| Class5+Class8 | 215 |
| Class1+Class9 | 213 |
| Class2+Class5 | 206 |
| Class1 | 202 |
| Class3+Class7 | 182 |
| Class5 | 173 |
| Class5+Class9 | 111 |

The total number of images is 5,995.

The group with the highest number of samples contains 1,756 images, while the group with the lowest number of samples contains 111 images.

### My Question

Does this level of data imbalance pose a significant problem during multi-class classification training with LightlyTrain?

Do you have any recommendations for this situation?

Thank you.

**Comment by liopeer** [comment]
Hi @enescicek,

I personally would classify this as not super imbalanced. My recommendation would be to make your test/validation splits large enough so that you're sure to spot issues if they arrive. If you use a strong backbone, something like DINOv2/DINOv3 or similar, I would definitely also try freezing the backbone during training, a very light-weight classifier on top of a frozen backbone is a really good way to remedy class imbalance. If you want to e.g. sub-sample a more balanced subset as e.g. your validation set I would recommend checking our [LightlyStudio](https://github.com/lightly-ai/lightly-studio). Currently we unfortunately do not support weighting the loss by class, however it's definitely something that we should add.

One question though: You're saying _multi-class_, i.e. are you really planning to map _Class3+Class7_ to a single class? Or do you mean that you will actually train a _multi-label_ model through the `classification_task="multilabel"` argument – because this is what I would recommend you do.

**Comment by enescicek** [comment]
Thank you for your response.Yes, I'll use classification_task="multilabel".
